src/libs/sfdk/emulator.cpp

```diff
@@ -442,6 +442,14 @@ void EmulatorPrivate::updateVmProperties(const QObject *context, const Functor<b
     });
 }
 
+bool EmulatorPrivate::isValid() const
+{
+    QTC_ASSERT(!sharedConfigPath.isEmpty(), return false);
+    QTC_ASSERT(!sharedSshPath.isEmpty(), return false);
+    QTC_ASSERT(virtualMachine->sshParameters().port(), return false);
+    return true;
```

Comment:
Similarly as with engine, the other properties like mac, subnet, ... should be checked too I think.